aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -248,22 +209,29 @@ static Tensor& sum_out(Tensor& result, const Tensor& self, IntArrayRef dim,
   return result;
 }
 
-static Tensor sum(const Tensor& self, IntArrayRef dim, bool keepdim, optional<ScalarType> dtype) {
+Tensor sum(const Tensor &self, c10::optional<ScalarType> dtype) {
+  return at::native::sum(self, {}, false, dtype);
+}
+
+Tensor& sum_out(Tensor& result, const Tensor& self, IntArrayRef dim, c10::optional<ScalarType> dtype) {
+  return at::native::sum_out(result, self, dim, false, dtype);
+}
+
+Tensor sum(const Tensor& self, IntArrayRef dim, bool keepdim, c10::optional<ScalarType> dtype) {
   Tensor result;
-  native::sum_out(result, self, dim, keepdim, dtype);
-  return result;
+  return at::native::sum_out(result, self, dim, keepdim, dtype);
 }
 
-Tensor sum(const Tensor &self, ScalarType dtype) {
-  return at::native::sum(self, {}, false, optional<ScalarType>(dtype));
+Tensor sum(const Tensor& self, IntArrayRef dim, bool keepdim) {
```

Comment:
do we need all these overloaded implementations?  I would expect we only need 3, which is the number of functions specified in native_functions, but there are 5 `sum` functions here.

Reply:
ah not sure why I had these, removed.